src/Arcus.Messaging.Pumps.Abstractions/Resiliency/DefaultMessagePumpCircuitBreaker.cs

```diff
@@ -95,33 +131,19 @@ protected MessagePump GetRegisteredMessagePump(string jobId)
 
         private async Task TryProcessSingleMessageAsync(MessagePump messagePump, MessagePumpCircuitBreakerOptions options)
         {
-            bool isRecovered = false;
-            while (!isRecovered)
-            {
-                MessageProcessingResult processingResult = await messagePump.TryProcessProcessSingleMessageAsync(options);
-                isRecovered = processingResult.IsSuccessful;
-
-                if (isRecovered)
-                {
-                    _logger.LogTrace("Message pump '{JobId}' successfully handled a single message, closing circuit...", messagePump.JobId);
-                    await messagePump.StartProcessingMessagesAsync(CancellationToken.None);
-                }
-                else
-                {
-                    await WaitMessageIntervalAsync(messagePump, options);
-                }
-            }
+            _logger.LogDebug("Try to process single message in message pump '{JobId}' (state: half-open)", messagePump.JobId);
+            await messagePump.TryProcessProcessSingleMessageAsync(options);
```

Comment:
This `TryProcessSingleMessageAsync` method of the message-pump retuns an indication whether or not the processing succeeded, shouldn't we return that value here in this private method as well and then do something with it ?

Reply:
Done here https://github.com/arcus-azure/arcus.messaging/pull/444/commits/23520db71ecf7a185ff911e967515de245d1656c